### PR TITLE
Revenant Micro Patch

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

- Revenants now have max 60 esscence at the start.
- Revenants can now break windows again.
- Revenants lose their regen when their ashes are tossed. They need to absorb a soul to regain it.
- Revenants have a random ability re-locked when their ashes are tossed.
- Tossing ashes now cripple revenants until they can regain their powers.

## Why It's Good For The Game

I am once again trying to balance revenant as an experiment.

## Changelog

:cl:
balance: Revenants now have max 60 esscence at the start.
balance: Revenants can now break windows again.
balance: Revenants lose their regen when their ashes are tossed. They need to absorb a soul to regain it.
balance: Revenants have a random ability re-locked when their ashes are tossed.
balance: Tossing ashes now cripple revenants until they can regain their powers.
balance: Revenants now reform after 15 seconds. 
/:cl:

### CODE REVIEW

File: modular_zubbers/code/modules/antagonists/revenant/revenant.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+/mob/living/basic/revenant/get_status_tab_items()
+	. = ..()
+	. += "Times scattered: [times_tossed]"
+
+
+/obj/item/ectoplasm/revenant/proc/become_defeated()
+	alpha = 0
+	revenant.times_tossed++
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/antagonists/revenant/revenant.dm:15]: Why are we tracking this when it has no effect

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/modules/antagonists/revenant/revenant.dm (hunk 1 above)
Comment (Majkl-J) [reply modular_zubbers/code/modules/antagonists/revenant/revenant.dm:15]: Also I don't see the alpha being reset anywhere

**Comment by YakumoChen** [comment]
How is this good for the game? It just makes revanents unkillable (they are already very hard to kill at times) and easily able to vent the entire station

**Comment by nevimer** [comment]
> How is this good for the game? It just makes revanents unkillable (they are already very hard to kill at times) and easily able to vent the entire station

1. They can't spam abilities due to super slow regen now
2. the higher impact abilitites can't be spammed at all, unless you have absorbed enough perfect souls now to use them. You will never regen enough essence to spam blight or EMP, unless you absorbed enough bodies with minds.
3. Once you unlock those abilities, using them puts you at single digit health numbers. Literally anything will make you ashes.
4. Most ability usage leaves you with single digit HP numbers and your regeneration is extremely slow now.
5. Being tossed means your regen might aswell not exist, until you absorb another body, and it won't be easy because you will have single digit health for a long while after being tossed.
6. Because being tossed doesn't permanently kill you, it will majorly inconvenience you unless you have people supporting you. This also means that you can afford much more risk knowing that you can't be permakilled by being tossed - allowing you to show yourself more and do riskier actions - and more opportunity for people to fight you conversely.

**Comment by YakumoChen** [comment]
> > How is this good for the game? It just makes revanents unkillable (they are already very hard to kill at times) and easily able to vent the entire station
> 
>     1. They can't spam abilities due to super slow regen now
> 
>     2. the higher impact abilitites can't be spammed at all, unless you have absorbed enough perfect souls now to use them. You will never regen enough essence to spam blight or EMP, unless you absorbed enough bodies with minds.
> 
>     3. Once you unlock those abilities, using them puts you at single digit health numbers. Literally anything will make you ashes.
> 
>     4. Most ability usage leaves you with single digit HP numbers and your regeneration is extremely slow now.
> 
>     5. Being tossed means your regen might aswell not exist, until you absorb another body, and it won't be easy because you will have single digit health for a long while after being tossed.
> 
>     6. Because being tossed doesn't permanently kill you, it will majorly inconvenience you unless you have people supporting you. This also means that you can afford much more risk knowing that you can't be permakilled by being tossed - allowing you to show yourself more and do riskier actions - and more opportunity for people to fight you conversely.

the changes sound like you're making it unfun for everyone. There's already a cap on how much essence you can absorb from monkies and mindless mobs, and absorbing perfect souls is naturally extremely hard, and completely dependent on how busy the round is with other antags.

**Comment by nevimer** [comment]
> > > How is this good for the game? It just makes revanents unkillable (they are already very hard to kill at times) and easily able to vent the entire station
> > 
> > 
> > ```
> > 1. They can't spam abilities due to super slow regen now
> > 
> > 2. the higher impact abilitites can't be spammed at all, unless you have absorbed enough perfect souls now to use them. You will never regen enough essence to spam blight or EMP, unless you absorbed enough bodies with minds.
> > 
> > 3. Once you unlock those abilities, using them puts you at single digit health numbers. Literally anything will make you ashes.
> > 
> > 4. Most ability usage leaves you with single digit HP numbers and your regeneration is extremely slow now.
> > 
> > 5. Being tossed means your regen might aswell not exist, until you absorb another body, and it won't be easy because you will have single digit health for a long while after being tossed.
> > 
> > 6. Because being tossed doesn't permanently kill you, it will majorly inconvenience you unless you have people supporting you. This also means that you can afford much more risk knowing that you can't be permakilled by being tossed - allowing you to show yourself more and do riskier actions - and more opportunity for people to fight you conversely.
> > ```
> 
> the changes sound like you're making it unfun for everyone. There's already a cap on how much essence you can absorb from monkies and mindless mobs, and absorbing perfect souls is naturally extremely hard, and completely dependent on how busy the round is with other antags.
> 
> Making it so you die in one hit to anything with terrible regen means the only real way to play the game is to break all the windows when nobody is around and avoid using powers on crew. This just discourages revanent gameplay while making sure they can't be removed fully from the round to keep breaking windows to grief crew.
> 
> Re-locking abilities more or just completely locks rev out of gameplay once they die once or twice since you can't regain the essence from mindless mobs.

I don't think you are understanding the changes. I can't help you with that.

**Comment by YakumoChen** [comment]
I do understand it just fine, watching someone play revanent with the new changes is pure suffering. They've died a few times so they're essentially locked out of doing anything but absorb and transmit

**Comment by NargaInDisguise** [comment]
This is an absolutely god awful change, unkillable rev makes the round miserable for both sides.

**Comment by SapphoQueer** [comment]
Just played through a shift with this. The unkillable-ness is the only part that sucks major-time. The rest is kinda neat. Maybe have a limit to the amount of tosses they can survive? I.E. first toss makes the "aborb a soul" thing kick in, second one is dead dead.

**Comment by NanoCats** [comment]
OK, I had a chance to demo this and was explicitly asked for my feedback so here goes: 

It's not fun. Everyone wants to kill you super badly and you want to kill them super badly, but neither party can kill the other. 

In the beginning, I can't absorb so much as the morgue bodies without access to Defile (I had to wait for Xenobio, which isn't a guarantee). I just have to kind of, pray for something to die that I can aborb somewhere. 

It's also trivial for literally anyone to rebuke me with salt, or just shooting me, because the core weakness of revenant is that you have to press a button that paralyzes you and locks you in place for more than long enough for someone to shoot you and then you have to dodge attacks at a super-slow-speed before you go back to being spectral. The captain literally stole the null rod and I couldn't do anything to him for the rest of the shift. 

Revenant gameplay, especially under this server, is "Pray/wait for someone to die, then get ass blasted the moment someone does and you try to absorb because nobody leaves a dead body unattended for longer than the five seconds that you need to absorb someone"  Reminder that last time I checked revenant health wasn't regular health, your essence is your health, and you have to use half of your health to use an ability, making you extremely vulnerable.

I targeted a synth character and used my EMP ability on them so many times it isn't funny.

Furthermore, people mentioned that venting the station was the only way to make ground, which I don't like because come on, even as an antag I try to avoid just casually venting the station for the lols.

Due to special testing-the-PR circumstances I was a revenant from the _start of the shift_ and I had a positively miserable, enraging time.

**Comment by nevimer** [comment]
> OK, I had a chance to demo this and was explicitly asked for my feedback so here goes: 
> 
> It's not fun. Everyone wants to kill you super badly and you want to kill them super badly, but neither party can kill the other. 
> 
> In the beginning, I can't absorb so much as the morgue bodies without access to Defile (I had to wait for Xenobio, which isn't a guarantee). I just have to kind of, pray for something to die that I can aborb somewhere. 
> 
> It's also trivial for literally anyone to rebuke me with salt, or just shooting me, because the core weakness of revenant is that you have to press a button that paralyzes you and locks you in place for more than long enough for someone to shoot you and then you have to dodge attacks at a super-slow-speed before you go back to being spectral. The captain literally stole the null rod and I couldn't do anything to him for the rest of the shift. 
> 
> Revenant gameplay, especially under this server, is "Pray/wait for someone to die, then get ass blasted the moment someone does and you try to absorb because nobody leaves a dead body unattended for longer than the five seconds that you need to absorb someone"  Reminder that last time I checked revenant health wasn't regular health, your essence is your health, and you have to use half of your health to use an ability, making you extremely vulnerable.
> 
> I targeted a synth character and used my EMP ability on them so many times it isn't funny.
> 
> Furthermore, people mentioned that venting the station was the only way to make ground, which I don't like because come on, even as an antag I try to avoid just casually venting the station for the lols.
> 
> Due to special testing-the-PR circumstances I was a revenant from the _start of the shift_ and I had a positively miserable, enraging time.

Under normal circumstances, the revenant spawns next to a body that will give enough esscence to unlock defile. I spawned you in that round, not next to a body. That's not relevant to the PR. 

There is only one Null rod, so that's not a big deal and also not relevant to the PR. 

You also can't vent the station at all due to the airbag mechanic. All broken windows get replaced with an airbag. If someone else doesn't replace the airbag when cleaning up your mess, only then can you do that. Or you can spam the emp hoping for the low emag chance on the door. That's also not relevant to the PR, the only difference is the PR actually lets you break the glass instead of make cracks.

Your experience sounds basically like how Revenant gameplay is - identical to the before and after this pr. 
If this PR did not exist, your round would have gone the same.

**Comment by NargaInDisguise** [comment]
> Under normal circumstances, the revenant spawns next to a body that will give enough esscence to unlock defile. I spawned you in that round, not next to a body. That's not relevant to the PR.
> 
> There is only one Null rod, so that's not a big deal and also not relevant to the PR.
> 
> You also can't vent the station at all due to the airbag mechanic. All broken windows get replaced with an airbag. If someone else doesn't replace the airbag when cleaning up your mess, only then can you do that. Or you can spam the emp hoping for the low emag chance on the door. That's also not relevant to the PR, the only difference is the PR actually lets you break the glass instead of make cracks.
> 
> Your experience sounds basically like how Revenant gameplay is - identical to the before and after this pr. If this PR did not exist, your round would have gone the same.

If this PR didn't exist, the revenant would have died and both parties would have suffered less grief. A revenant who can barely function and a crew who has to live with that looming threat of a threat that literally cannot be killed under any circumstances. You were told that making it nigh impossible for anyone besides JUST the chaplain able to scatter the ashes was a terrible idea, this is considerably worse.

**Comment by nikothedude** [comment]
why is this still merged? in no world is an unkillable revenant a good idea wtf

**Review by Swiftfeather — CHANGES_REQUESTED** [review]
Having played rev the other night and not knowing about this pr I wasn't aware anything had been changed, but I should have absolutely died at least 10 times and even said "it's so joever" multiple times when I SHOULD have died. 

The regen change is unnecessary and annoying. 

A few rounds prior there was a rev I thought was I killed for real but it turned out no, it was not dead. I thought another entire rev had mysteriously spawned in the same round but now I see it was the same one that should have been dead. They proceeded to bust atmos cans and incinerate atmos and all the effort that had been put into the project, even though they should have already been dead because what I had done should have killed them.

**Comment by nevimer** [comment]
Back at it again with more tweaks

**Comment by NargaInDisguise** [comment]
> Back at it again with more tweaks

Please do not bring this back. It was universally agreed to be a horrible idea to let a Revenant come back to life under any circumstances.

**Comment by nikothedude** [comment]
please just keep this PR closed _noone_ likes it

**Comment by NargaInDisguise** [comment]
Once again begging for this to be closed because its incredibly unhealthy to have revenants fucking respawn themselves no matter how big of a downside they get.

**Comment by nikothedude** [comment]
it baffles me as to how this PR remains test-merged despite the incredibly negative response

**Comment by buffyuwu** [comment]
Revenants were annoying before this, they're unkillable with this. Please for the love of god do not buff revenants as an 'experiment'.

**Review by Majkl-J — CHANGES_REQUESTED** [review]
I don't think making any self-revive antags unable to be RRd in some capacity is healthy for the game. An uncontainable antag should be able to be defeated

**Comment by Boviro** [comment]
I don't see a reason why this has to have been testmerged for approaching 2 months. Revenants do not like it, crew do not like it. It is not a fun experience for all involved. I agree that revenants need some rework but making one of the more annoying antags completely unkillable while also kneecapping them isn't healthy for the player experience.

**Comment by Cyprex** [comment]
I really don't think this is a good idea.

**Pull request closed without merging**